src/workerd/util/mimetype.c++

```diff
@@ -6,24 +6,64 @@
 #include <kj/debug.h>
 #include <kj/string-tree.h>
 #include <workerd/util/string-buffer.h>
+#include <array>
 
 namespace workerd {
 
 namespace {
 
-bool isWhitespace(const char c) {
+constexpr bool isWhitespace(const char c) noexcept {
   return (c == '\r' || c == '\n' || c == '\t' || c == ' ');
 }
 
-bool isTokenChar(const unsigned char c) {
-  return c == '!' || c == '#' || c == '$' || c == '%' || c == '&' || c == '\'' ||
-         c == '*' || c == '+' || c == '\\' || c == '-' || c == '.' || c == '^' ||
-         c == '_' || c == '`' || c == '|' || c == '~' || (c >= 'A' && c <= 'Z') ||
-         (c >= 'a' && c <= 'z') || (c >= '0' && c <= '9');
+constexpr std::array<uint8_t, 256> token_table = []() constexpr {
```

Comment:
I didn't know you can have constexpr lambda (which seems logical in retrospect). This is cool!